src/engine/Algebra.ts

```diff
@@ -68,11 +71,25 @@ module ex {
          return true;
       }
 
+      constructor();
+      /**
+       * @param vectorOptions coordindates of the Vector
+       */
+      constructor(vectorOptions: IVectorOptions);
       /**
        * @param x  X component of the Vector
        * @param y  Y component of the Vector
        */
-      constructor(public x: number, public y: number) { }
+      constructor(x: number, y: number);
+      constructor(optionsOrX?: number | IVectorOptions, y?: number) {
+            if (typeof optionsOrX === 'number') {
```

Comment:
Think you meant 'classes' not 'actors', but this makes a lot of sense and I will plan on this. I must have been tired because I had my in head that this would be an issue for existing constructors where there was just one arguments that was an object, but we would of course just not change those.